osbenchmark/builder/configs/utils/config_path_resolver.py

```diff
@@ -0,0 +1,38 @@
+import os
+
+from osbenchmark.exceptions import SystemSetupError
+from osbenchmark.utils.repo import BenchmarkRepository
+
+
+class ConfigPathResolver:
+    def __init__(self, cfg):
+        self.cfg = cfg
+
+    def resolve_config_path(self, config_type, config_version):
+        config_root_path = self._get_config_root_path()
+        root_path = os.path.join(config_root_path, config_type, f"v{config_version}")
+        if not os.path.exists(root_path):
+            raise SystemSetupError(f"Path {root_path} for {config_type} does not exist.")
+
+        return root_path
+
+    def _get_config_root_path(self):
+        root_path = self.cfg.opts("builder", "provision_config.path", mandatory=False)
+        if root_path:
+            return root_path
+        else:
+            distribution_version = self.cfg.opts("builder", "distribution.version", mandatory=False)
```

Comment:
Nit: Seems like terms such as "builder", "provision_configs" are repeated in a variety of places. Might be better to place them in an enum like what you did with "opensearch" references.

https://github.com/opensearch-project/opensearch-benchmark/pull/183